src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1231,6 +1239,11 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         {
             _cursorTimer.value().Start();
         }
+
+        if (!SystemParametersInfo(SPI_GETWHEELSCROLLLINES, 0, &_systemRowsToScroll, 0))

```

Comment:
Is calling this every time the control gets focused expensive? Seems to me that it might be easier for us to query this value once, when we're parsing the value originally, and just pass that value down. Then, we wouldn't need to keep `_systemRowsToScroll` around - we could just rely on the `RowsToScroll` value passed to us.

Reply:
It's not expensive. I initially had it in the scroll handler and it didn't seem to have adverse effects there, but I thought that was rather unneccesary processing so I moved it to the focus received handler. Putting it on the focus handler means that the app doesn't need re-initialization when you change the system value. If you put it in the initial parsing, then you'd have to reinitialize the settings to make changes take effect, which is not what happens in other apps.

Reply:
I've changed it now so that the globablappsettings will assign 0 to the internal value if anything other than a non-zero int value is in the settings. The terminal settings will then in the accessor determine if a valid setting is provided, and if not return the system setting. The terminal control will update it's internal value using that accessor when settings change and when it receives focus so that you don't have to restart the terminal when you change the system setting.

Reply:
Okay this makes sense, thanks for the explanation!